lib/src/test/kotlin/starknet/utils/DevnetClient.kt

```diff
@@ -1,19 +1,34 @@
 package starknet.utils
 
 import com.swmansion.starknet.data.types.Felt
+import com.swmansion.starknet.service.http.HttpService
+import com.swmansion.starknet.service.http.OkhttpHttpService
 import kotlinx.serialization.SerialName
 import kotlinx.serialization.Serializable
 import kotlinx.serialization.json.Json
+import java.lang.Exception
 import java.nio.file.Path
+import java.nio.file.Paths
 import java.util.concurrent.TimeUnit
 import kotlin.io.path.absolutePathString
+import kotlin.io.path.exists
 
-class DevnetClient(val host: String = "0.0.0.0", val port: Int = 5050) : AutoCloseable {
-    private var devnetProcess: Process? = null
+class DevnetClient(
+    private val host: String = "0.0.0.0",
+    private val port: Int = 5050,
+    private val httpService: HttpService = OkhttpHttpService(),
+) : AutoCloseable {
+    private val accountDirectory = Paths.get("src/test/resources/account")
+    private val baseUrl: String = "http://$host:$port"
 
-    val gatewayUrl: String
-    val feederGatewayUrl: String
-    val rpcUrl: String
+    private lateinit var accountAddress: Felt
+    private lateinit var devnetProcess: Process
+
+    private var isDevnetRunning = false
```

Comment:
Why do we need it?

Reply:
Mostly case you tried to do `.start()` multiple times, to make sure we don't spawn multiple devnet processes.

Reply:
Is this possible though? I think invoking `.start()` should fail, since the port is already in use (although let's verify that)

Reply:
It should fail but I think it's a little bit safer this way.